src/wp-includes/option.php

```diff
@@ -776,6 +776,24 @@ function update_option( $option, $value, $autoload = null ) {
 	 */
 	$value = apply_filters( 'pre_update_option', $value, $option, $old_value );
 
+	/*
+	 * To get the actual raw old value from the database, any existing pre filters need to be temporarily disabled.
+	 * Immediately after getting the raw value, they are reinstated.
+	 * The raw value is only used to determine whether a value is present in the database. It is not used anywhere
+	 * else, and is not passed to any of the hooks either.
+	 */
+	if ( has_filter( "pre_option_{$option}" ) ) {
+		global $wp_filter;
+
+		$old_filters = $wp_filter[ "pre_option_{$option}" ];
+		unset( $wp_filter[ "pre_option_{$option}" ] );
```

Comment:
```suggestion
		remove_all_filters( "pre_option_{$option}" );
```
Could we use this?

Reply:
@spacedmonkey I tried it, but the problem is that it also calls the `remove_all_filters()` method on the `WP_Hook` object in `$wp_filter[ "pre_option_{$option}" ]`. So we then cannot reinstate the filters that were previously there.